src/main/java/hudson/plugins/scm/koji/KojiSCM.java

```diff
@@ -144,18 +143,19 @@ public void checkout(Run<?, ?> run, Launcher launcher, FilePath workspace, TaskL
         }
 
         // TODO add some flag to allow checkout on local or remote machine
-        KojiBuildDownloader downloadWorker = new KojiBuildDownloader(createConfig(), createNotProcessedNvrPredicate
-                (run.getParent()), run.getCause(Cause.UserIdCause.class) != null);
+        KojiBuildDownloader downloadWorker = new KojiBuildDownloader(createConfig(),
+                                                                     createNotProcessedNvrPredicate(run.getParent()),
+                                                                     run.getParent());
         downloadWorker.setListener(listener);
-        KojiBuildDownloadResult downloadResult = workspace.act(downloadWorker);
+        KojiBuildDownloadResult downloadResult = downloadWorker.invoke(new File(workspace.toURI()), null);
```

Comment:
This seems to be the only real-changing line. I guess yiu tested it on remote host?
I would be expecting some change from file to url, but nothing like that arround.
How does it work?  invoke got the bits form master via serialization?

Reply:
this was wrong. Please see latest changes.

Reply:
also this is second really important change https://github.com/judovana/jenkins-scm-koji-plugin/pull/26/files?diff=unified#diff-11be36ffb70e08d83fd8f4774648ac3eR214